src/redux/reducers/inventory/entities.js

```diff
@@ -0,0 +1,69 @@
+import { ACTION_TYPES, SELECT_ENTITY, CHANGE_SORT, FILTER_ENTITIES } from '../../action-types';
+import { mergeArraysByKey } from '../../../Utilities/helpers';
+import { SortDirection } from '../../../PresentationalComponents/Table';
+import get from 'lodash/get';
+import orderBy from 'lodash/orderBy';
+
+export const defaultState = { loaded: false };
+
+const defaultColumns = [
+  {key: 'display_name', title: 'Name', composed: ['facts.release', 'display_name']},
+  {key: 'facts.last_seen', title: 'Last Seen'}
+]
+
+function entitiesPending(state) {
+    return {
+        ...state,
+        columns: mergeArraysByKey([state.columns, defaultColumns], 'key'),
+        loaded: false
+    };
+}
+
+function filterEntities(state, {payload: {key, filterString}}) {
+    const entities = filterString ?
+      state.rows.filter(item => item[key] && item[key].indexOf(filterString) !== -1) :
+      [...state.rows];
+    return {
+        ...state,
+        entities: entities
+    }
+}
+
+function entitiesLoaded(state, { payload }) {
+    const entities = mergeArraysByKey([state.rows, payload]);
+    return {
+        ...state,
+        loaded: true,
+        rows: entities,
```

Comment:
@karelhala : why do you have `rows: entities` here and also `entities` below?

Reply:
Rows represent data receives from db, but entities represent filtered and/or sorted data in which connected component is really interested.